tests/fixtures/optimizer/simplify.sql

```diff
@@ -602,3 +602,33 @@ TRUE;
 
 x = 2018 OR x <> 2018;
 x <> 2018 OR x = 2018;
+
+--------------------------------------
+-- Coalesce
+--------------------------------------
+COALESCE(x);
+x;
+
+COALESCE(x, 1) = 2;
+x = 2 AND NOT x IS NULL;
```

Comment:
why do we need the null check?

Reply:
I think the `null` will bubble up without this check and you may get different results if you only keep `x = 2`:

```
georgesittas=# with cte(x) as (select 1 union all select null) select coalesce(x, 1) = 2 from cte;
 ?column?
----------
 f
 f
(2 rows)

georgesittas=# with cte(x) as (select 1 union all select null) select x = 2 from cte;
 ?column?
----------
 f

(2 rows)

georgesittas=# with cte(x) as (select 1 union all select null) select x = 2 and not x is null from cte;
 ?column?
----------
 f
 f
(2 rows)
```

Reply:
```
COALESCE(NULL, 1) = 2 == FALSE
NULL = 2 == NULL
NULL = 2 AND NOT NULL IS NULL = FALSE
```